liblwgeom/lwgeom_transform.c

```diff
@@ -172,3 +205,238 @@ lwproj_from_string(const char *str1)
 	}
 	return pj_init_plus(str1);
 }
+
+/***************************************************************************/
+
+#else /* POSTGIS_PROJ_VERION >= 60 */
+
+int
+lwgeom_transform_from_str(LWGEOM *geom, const char* instr, const char* outstr)
+{
+	PJ *pj = proj_create_crs_to_crs(NULL, instr, outstr, NULL);
+	if (!pj)
+	{
+		PJ *pj_in = proj_create(NULL, instr);
+		PJ *pj_out = proj_create(NULL, outstr);
+		if (!pj_in)
+			lwerror("could not parse proj string '%s'", instr);
+		if (!pj_out)
+			lwerror("could not parse proj string '%s'", outstr);
+		return LW_FAILURE;
+	}
+
+	return lwgeom_transform(geom, pj);
+}
+
+int
+lwgeom_transform(LWGEOM* geom, PJ* pj)
+{
+	uint32_t i;
+
+	/* No points to transform in an empty! */
+	if (lwgeom_is_empty(geom))
```

Comment:
Doesn't it make more sense to check this before calling `lwgeom_transform` to avoid all the work involved in `GetPJUsingFCInfo`?

Reply:
Mmm, I like the cleanliness of the check here...